tedge/src/cli/mqtt/publish.rs

```diff
@@ -0,0 +1,75 @@
+use crate::cli::mqtt::MqttError;
+use crate::command::{Command, ExecutionContext};
+use futures::future::FutureExt;
+use mqtt_client::{Client, Message, QoS, Topic};
+use std::time::Duration;
+use tokio::select;
+
+pub struct MqttPublishCommand {
+    pub topic: String,
+    pub message: String,
+    pub qos: QoS,
+    pub mqtt_config: mqtt_client::Config,
+    pub client_id: String,
+    pub disconnect_timeout: Duration,
+}
+
+impl Command for MqttPublishCommand {
+    fn description(&self) -> String {
+        format!(
+            "publish the message \"{}\" on the topic \"{}\" with QoS \"{:?}\".",
+            self.message, self.topic, self.qos
+        )
+    }
+
+    fn execute(&self, _context: &ExecutionContext) -> Result<(), anyhow::Error> {
+        Ok(publish(self)?)
+    }
+}
+
+#[tokio::main]
+async fn publish(cmd: &MqttPublishCommand) -> Result<(), MqttError> {
+    let mut mqtt = cmd
+        .mqtt_config
+        .clone()
+        .connect(cmd.client_id.as_str())
+        .await?;
+
+    let tpc = Topic::new(cmd.topic.as_str())?;
```

Comment:
change varible name to `topic`. Also think about changing the type of `topic` in MqttPublishCommand from `String` to `Topic`.